src/Webcreate/Vcs/Git/Parser/CliParser.php

```diff
@@ -155,11 +155,15 @@ public function parseDiffOutput($output, array $arguments = array())
 
         $retval = array();
         foreach ($lines as $line) {
-            if (preg_match('/([ACDMRTUXB])\d*\s+(.*)?/', $line, $matches)) {
-                list($fullmatch, $x, $file) = $matches;
+            if (preg_match('/([ACDMRTUXB])(\d*)\s+(.*)?/', $line, $matches)) {
+                list($fullmatch, $status, $number, $filename) = $matches;
 
-                $file = new VcsFileInfo($file, $this->getClient()->getHead());
-                $file->setStatus($x);
+                if ($number && preg_match('/(.*\s)+(.*)/', $filename, $filenameMatches)) {
```

Comment:
I think this might fail for filenames with spaces, but it's good enough for now.